src/System.Private.Xml/src/System.Private.Xml.csproj

```diff
@@ -7,7 +7,9 @@
     <RootNamespace>System.Xml</RootNamespace>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <NoWarn>$(NoWarn),649</NoWarn>
-    <DefineConstants Condition="'$(TargetGroup)' == 'uap'">$(DefineConstants);uap</DefineConstants>
+    <FeatureCompiledXsl Condition="'$(TargetGroup)' != 'uap' AND '$(TargetGroup)' != 'uapaot'">true</FeatureCompiledXsl>
+    <DefineConstants Condition="'$(FeatureCompiledXsl)' == 'true'">$(DefineConstants);FEATURE_COMPILED_XSL</DefineConstants>
+    <DefineConstants Condition="'$(TargetGroup)' == 'uapaot'">$(DefineConstants);FEATURE_SERIALIZATION_UAPAOT</DefineConstants>
```

Comment:
@jkotas I do not have much context on serialization code and I've given similar argument to @shmao when we talked about how to name this but from our conversation it seems that this implementation is simply AOT specific. I wouldn't spend too much time on trying to figure out better name since all owners know what it's for and we have achieved the purpose of the switch of having one centralize place to change logic.